kodilla-hibernate/src/test/java/com/kodilla/hibernate/tasklist/dao/TaskListDaoTestSuite.java

```diff
@@ -39,4 +42,31 @@ public void testFindByListName() {
         taskListDao.delete(id2);
     }
 
+    @Test
+    public void testTaskListDaoSaveWithTasks() {
+        //Given
+        Task task = new Task("Test: Learn Hibernate", 14);
+        Task task2 = new Task("Test: Write some entities", 3);
+        TaskFinancialDetails tfd = new TaskFinancialDetails(new BigDecimal(20), false);
+        TaskFinancialDetails tfd2 = new TaskFinancialDetails(new BigDecimal(10), false);
+        task.setTaskFinancialDetails(tfd);
+        task2.setTaskFinancialDetails(tfd2);
+        TaskList taskList = new TaskList(LIST_NAME, "ToDo tasks");
+        taskList.getTasks().add(task);
+        taskList.getTasks().add(task2);
+        task.setTaskList(taskList);
+        task2.setTaskList(taskList);
+
+        //When
+        taskListDao.save(taskList);
```

Comment:
przed save na taskList musisz zrobic save na produktach
`productDao.save(product1);` i pozostale

Reply:
* wazne - przed, nie po. Inaczej dostaniesz exception. Mozemy to omowic na konsultajach

Reply:
```
    @Test
    public void testInvoiceDaoSave() {
        //Given
        Product product1 = new Product("Product 1");
        Product product2 = new Product("Product 2");
        Product product3 = new Product("Product 3");
        productDao.save(product1);
        productDao.save(product2);
        productDao.save(product3);

        Item item1 = new Item(new BigDecimal("300.00"), 20);
        Item item2 = new Item(new BigDecimal("300.00"), 50);
        Item item3 = new Item(new BigDecimal("5000.00"), 3);
        Item item4 = new Item(new BigDecimal("20.00"), 200);
        Item item5 = new Item(new BigDecimal("20.00"), 100);

        Invoice invoice = new Invoice("1234567899");

        invoice.getItems().add(item1);
        invoice.getItems().add(item2);
        invoice.getItems().add(item3);
        invoice.getItems().add(item4);
        invoice.getItems().add(item5);

        item1.setInvoice(invoice);
        item2.setInvoice(invoice);
        item3.setInvoice(invoice);
        item4.setInvoice(invoice);
        item5.setInvoice(invoice);

        item1.setProduct(product1);
        item2.setProduct(product1);
        item3.setProduct(product2);
        item4.setProduct(product3);
        item5.setProduct(product3);

        product1.getItems().add(item1);
        product1.getItems().add(item2);
        product2.getItems().add(item3);
        product3.getItems().add(item4);
        product3.getItems().add(item5);
        
        //When
        invoiceDao.save(invoice);
        int id = invoice.getId();

        //Then
        Assert.assertNotEquals(0, id);

        //CleanUp
    }

Reply:
Nadal mi nie działa.  Mam exception:

```org.springframework.orm.jpa.JpaSystemException: could not execute statement; nested exception is org.hibernate.exception.GenericJDBCException: could not execute statement```
(...)
```Caused by: java.sql.SQLException: Field 'product_name' doesn't have a default value```

Tak wygląda moja tabela w workbench:
![obraz](https://user-images.githubusercontent.com/33154581/34467335-c32193fc-eeee-11e7-8d4f-c8ea9c0c9188.png)

A w klasie Produkt mam tylko dwa pola, które powinny zostać utrwalone w kolumnach
```@Id
    @GeneratedValue
    @NotNull
    @Column(name = "ID", unique = true)
    public int getId() {
        return id;
    }

    @NotNull
    @Column(name = "NAME")
    public String getName() {
        return name;
    }```

Możliwe, że zmieniałem nazwy kolumn w kodzie i dlatego dodały się dodatkowe kolumny "product_id" i "product_name". Czy to przez te kolumny pojawia się wyjątek? Jak to naprawić?